wordpress-mobile/WordPress-Android
# Issue/6926 site creation theme selection screen

Fixes #6926 

Introduces the Theme selection screen. The offered theme options depend on which site category was chosen on the previous wizard step.

Architecture details:
* A headless fragment (`SiteCreationThemeLoaderFragment`) is used for triggering the WPCOM themes fetch and it's invoked early in the wizard process so the result to be available soon.
* Rotation doesn't cause a new network call for the themes since the headless loader fragment is retained.
* `SiteCreationThemeFragment` is listening to the sticky event emitted from the loader fragment to populate the list. If data are available upon entering the themes screen then no progress indicator is ever shown and the list gets populated immediately. Otherwise, some waiting (with progress bar visible) will happen.

To test:
* With the app logged in to a wpcom account, head over to MySite > Switch Site, tap on the "+" icon in the top toolbar and tap on "Create WordPress.com site"
* Tap on one of the categories
* Notice the list of themes. Should be different for each category (at the time of writing).
* Tap on a theme and notice the Toast with the theme-id popping up.
